src/main/scala/lms/transformation/distributed_tensor/distributedTensorMisc.scala

```diff
@@ -18,18 +18,51 @@ trait FixedSizeDistributedTensorMiscTypeLess extends FixedSizeDistributedTensorM
 
   def SoftmaxForward(input: TENSOR, params: SoftmaxParam, anno: Anno, __pos: SourceContext): TENSOR = {
     val res_tt = input.resultType
-    (new TENSOR(Adapter.g.reflectRead("tensor_softmax", C(res_tt), C(anno), input.x, 
+    (new TENSOR(Adapter.g.reflectRead("tensor_softmax", C(res_tt), C(anno), input.x,
       C(params))(input.x)).withSrcType(__pos, input.et))
   }
 
   def SoftmaxBackward(output: TENSOR, doutput: TENSOR, params: SoftmaxParam, anno: Anno, __pos: SourceContext): TENSOR = {
     val res_tt = doutput.resultType
-    (new TENSOR(Adapter.g.reflectRead("tensor_softmax_bwd", C(res_tt), C(anno), output.x, doutput.x, 
+    (new TENSOR(Adapter.g.reflectRead("tensor_softmax_bwd", C(res_tt), C(anno), output.x, doutput.x,
       C(params))(output.x, doutput.x)).withSrcType(__pos, doutput.et))
   }
 
+  def MaskedFillForward[T](input: TENSOR, mask: TENSOR, value: Float, anno: Anno, __pos: SourceContext): TENSOR = {
+    // require(input.shapeSize == mask.shapeSize, "shape of mask must be equal to shape of the tensor")
```

Comment:
maybe we need to have this `require` for now? I think our template cannot support broadcasting yet.

Reply:
fixed in 121424f